### ISSUE TITLE
[Bug]: After update from 29 to 30, Your web server is not properly set up to resolve `.well-known` URLs

### ISSUE THREAD

**Issue by marco-calautti** [nextcloud/server#48232]
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on [Github](https://github.com/nextcloud/server/issues?q=is%3Aopen+is%3Aissue+label%3Abug) OR [Nextcloud Community Forum](https://help.nextcloud.com/) _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

After updating from NC29 to NC30 the report shows:
Your web server is not properly set up to resolve `.well-known` URLs, failed on: `/.well-known/caldav`

I tried accessing https://<myhost>/.well-known/caldav (and also carddav), and the redirect to remote.php works fine, but for some reasons NC30 is not able to verify that the redirect indeed works.

### Steps to reproduce

1. Update from NC29 to NC30
2. Go to administration -> Report

### Expected behavior

NC30 should be able to understand that the .well-known redirect works.

### Nextcloud Server version

30

### Operating system

None

### PHP engine version

None

### Web server

Apache (supported)

### Database engine version

MariaDB

### Is this bug present after an update or on a fresh install?

Upgraded to a MAJOR version (ex. 28 to 29)

### Are you using the Nextcloud Server Encryption module?

Encryption is Disabled

### What user-backends are you using?

- [X] Default user-backend _(database)_
- [ ] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

**Comment by joshtrichards** [comment]
Please provide the output of `curl -I -L https://<myhost>/.well-known/caldav` from the command-line of your Nextcloud Server.

**Comment by marco-calautti** [comment]
`HTTP/2 401 
content-security-policy: default-src 'none';base-uri 'none';manifest-src 'self';script-src 'self';style-src 'self' 'unsafe-inline';img-src 'self' data: blob:;font-src 'self' data:;connect-src 'self';media-src 'self';frame-ancestors 'self';form-action 'self'
content-type: application/xml; charset=utf-8
date: Fri, 20 Sep 2024 15:27:55 GMT
referrer-policy: no-referrer
server: Apache/2.4.62 (Debian)
set-cookie: oclmolb45e6m=5c351b3807e1056eaf31c5243c9f136f; path=/; secure; HttpOnly; SameSite=Lax
set-cookie: oc_sessionPassphrase=UeE1ZVSvIp%2F0yMYIN%2FIYGiNjuNIWcYLhEVE6xfeFI%2BqfuW9wkGKkgtD%2BxJn9TBbIDBcLLEXPictwyl8faFtEoYrn%2FHJ3smoXor90uYIX45pca6pwtHUy59hWLlczUcQG; path=/; secure; HttpOnly; SameSite=Lax
set-cookie: oclmolb45e6m=5c351b3807e1056eaf31c5243c9f136f; path=/; secure; HttpOnly; SameSite=Lax
set-cookie: __Host-nc_sameSiteCookielax=true; path=/; httponly;secure; expires=Fri, 31-Dec-2100 23:59:59 GMT; SameSite=lax
set-cookie: __Host-nc_sameSiteCookiestrict=true; path=/; httponly;secure; expires=Fri, 31-Dec-2100 23:59:59 GMT; SameSite=strict
set-cookie: oclmolb45e6m=5c351b3807e1056eaf31c5243c9f136f; path=/; secure; HttpOnly; SameSite=Lax
strict-transport-security: max-age=31536000; preload
www-authenticate: Basic realm="Nextcloud", charset="UTF-8"
x-content-type-options: nosniff
x-frame-options: SAMEORIGIN
x-permitted-cross-domain-policies: none
x-powered-by: PHP/8.2.23
x-robots-tag: noindex, nofollow
x-xss-protection: 1; mode=block
`

**Comment by joshtrichards** [comment]
Is that really the full output? No redirect beforehand to `/remote.php/dav/`?

**Comment by joshtrichards** [comment]
With this Traefik config? 

https://docs.nextcloud.com/server/latest/admin_manual/configuration_server/reverse_proxy_configuration.html#traefik-2

It sounds like you may be doing a rewrite rather than a redirect, which would explain the problem.

**Comment by marco-calautti** [comment]
Yes, I am using a rewrite, as also suggested by the official Nextcloud's documentation you reported:

`traefik.http.middlewares.nextcloud-dav.replacepathregex.regex: "^/.well-known/ca(l|rd)dav"`
`traefik.http.middlewares.nextcloud-dav.replacepathregex.replacement: "/remote.php/dav/"`

**Comment by marco-calautti** [comment]
After changing to a redirectregex, the warning disappeared. Thanks a lot! Very strange that with previous versions of Nextcloud, the problem did not occur.

**Comment by joshtrichards** [comment]
The setup checks have evolved quite a bit through the v28-v30 development cycles.
